File: uikit/src/main/java/io/agora/chat/uikit/utils/EaseVoiceLengthUtils.java
package io.agora.chat.uikit.utils;

import android.content.Context;

/**
 * use to control voice view's length
 */
public class EaseVoiceLengthUtils {
    /**
     * Get the length of the voice
     * @param context
     * @param voiceLength
     * @return
     */
    public static int getVoiceLength(Context context, int voiceLength) {
        float maxLength = EaseUtils.getScreenInfo(context)[0] / 4 - EaseUtils.dip2px(context, 10);
        float paddingLeft;
        if(voiceLength <= 20) {
            paddingLeft = voiceLength / 20f * maxLength + EaseUtils.dip2px(context, 10);
        }else {
            paddingLeft = maxLength + EaseUtils.dip2px(context, 10);
        }
        return (int) paddingLeft;
    }
}

File: ease-im-kit/src/main/kotlin/com/hyphenate/easeui/common/utils/EaseVoiceLengthUtils.kt
package com.hyphenate.easeui.common.utils

import android.content.Context
import com.hyphenate.easeui.common.extensions.dpToPx
import com.hyphenate.easeui.common.extensions.getScreenInfo

/**
 * use to control voice view's length
 */
object EaseVoiceLengthUtils {
    /**
     * Get the length of the voice
     * @param context
     * @param voiceLength
     * @return
     */
    fun getVoiceLength(context: Context, voiceLength: Int): Int {
        val maxLength: Float =
            context.getScreenInfo()[0] / 4 - 10.dpToPx(context)
        val paddingLeft: Float = if (voiceLength <= 20) {
            voiceLength / 20f * maxLength + 10.dpToPx(context)
        } else {
            maxLength + 10.dpToPx(context)
        }
        return paddingLeft.toInt()
    }
}